image/copy/single.go

```diff
@@ -645,13 +682,19 @@ func (ic *imageCopier) copyConfig(ctx context.Context, src types.Image) error {
 			return destInfo, nil
 		}()
 		if err != nil {
-			return err
+			return nil, err
 		}
 		if destInfo.Digest != srcInfo.Digest {
-			return fmt.Errorf("Internal error: copying uncompressed config blob %s changed digest to %s", srcInfo.Digest, destInfo.Digest)
+			// Allow digest algorithm changes when forcing a specific digest algorithm
+			forcingDifferentAlgo := ic.c.options.digestOptions.MustUseSet() != "" &&
```

Comment:
- See the similar check `copyBlobFromStream`
- “MustUse” Is not the only reason a copy might choose a different algorithm (e.g. we might be pushing to a sha256-only-transport or registry), hard-coding a check _only for `MustUse`_ here is overly specific
- Compare the end of `copyLayers`, where the equality check is _immediately paired_ with manifest edits [and see elsewhere about manifest edits], so that we always set up edits if we know they are needed, and we minimize the risk of the two getting out of sync. I’m not sure whether the check+update should live in `copyConfig` or `copyUpdatedConfigAndManifest` (guessing the latter), but they should be together.